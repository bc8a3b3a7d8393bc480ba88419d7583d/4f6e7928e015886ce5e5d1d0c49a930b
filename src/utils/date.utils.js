import moment from 'moment';

const supportedDateFormats = {
  ddddDDMMMYYYY: 'dddd, DD MMM YYYY',
};

const format = (date, dateFormat) => moment(date).format(dateFormat);

export default {
  format,
  supportedDateFormats,
};
