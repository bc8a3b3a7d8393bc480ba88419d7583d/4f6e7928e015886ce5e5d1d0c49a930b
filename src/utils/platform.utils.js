import { Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import { equals } from 'ramda';

const isDEV = () => equals(process.env.NODE_ENV, 'development');
const isIOS = () => equals(Platform.OS, 'ios');
const isAndroid = () => equals(Platform.OS, 'android');
const iPhoneX = 'iPhone X';
const isIPhoneX = () => isIOS() && equals(DeviceInfo.getModel(), iPhoneX);

export { isDEV, isIOS, isAndroid, isIPhoneX, iPhoneX };
