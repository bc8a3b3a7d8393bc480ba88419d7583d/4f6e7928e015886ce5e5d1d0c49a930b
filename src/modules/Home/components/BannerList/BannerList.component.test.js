import { testSnapshots } from '../../../../utils/test.utils';

import Component from './BannerList.component';

describe('Modules/Home - BannerList', () => {
  testSnapshots(Component, [
    {
      description: 'basic render',
      props: {
        onPress: jest.fn(),
      },
    },
  ]);

  test('check default props', () => {
    expect(Component.defaultProps).toEqual(
      expect.objectContaining({
        onPress: expect.any(Function),
      }),
    );
  });

  test('check return value of default function props', () => {
    const onPressFunctionResult = Component.defaultProps.onPress();
    expect(onPressFunctionResult).toEqual(undefined);
  });
});
