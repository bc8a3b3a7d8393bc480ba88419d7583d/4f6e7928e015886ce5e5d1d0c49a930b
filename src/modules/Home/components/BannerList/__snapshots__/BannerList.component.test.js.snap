// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modules/Home - BannerList snapshots basic render 1`] = `
<FlatList
  data={
    Array [
      Object {
        "attributes": Object {
          "highlighted": "false",
          "specialItemBack": "false",
          "specialItemFront": "false",
        },
        "countdown": null,
        "iconTitle": null,
        "items": Array [
          Object {
            "attributes": Object {
              "backgroundImage": "https://tvlk.imgix.net/imageResource/2018/10/29/1540785749153-f3fe687a48f52ebc64a754c1b16b4457.jpeg?auto=compress%2Cformat&cs=srgb&fm=pjpg&ixlib=java-1.1.12&q=75",
              "cornerLabelText": null,
              "iconImage": null,
              "ribbonText": null,
              "subtitle": null,
              "title": null,
            },
            "id": "traveloka://flight/search/one_way/0/HAN/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "link": "traveloka://flight/search/one_way/0/HAN/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "linkType": "DEEPLINK",
            "style": Object {
              "properties": Object {
                "additionalTextColor": "#e91d1d",
                "subtitleColor": "#8f8f8f",
                "titleColor": "#1b1b1b",
              },
            },
            "type": "default",
          },
          Object {
            "attributes": Object {
              "backgroundImage": "https://tvlk.imgix.net/imageResource/2018/10/29/1540783824593-6639dc81ec97d8309915d5b1f475b61f.jpeg?auto=compress%2Cformat&cs=srgb&fm=pjpg&ixlib=java-1.1.12&q=75",
              "cornerLabelText": null,
              "iconImage": null,
              "ribbonText": null,
              "subtitle": null,
              "title": null,
            },
            "id": "traveloka://flight/search/one_way/0/DAD/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "link": "traveloka://flight/search/one_way/0/DAD/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "linkType": "DEEPLINK",
            "style": Object {
              "properties": Object {
                "additionalTextColor": "#e91d1d",
                "subtitleColor": "#8f8f8f",
                "titleColor": "#1b1b1b",
              },
            },
            "type": "default",
          },
          Object {
            "attributes": Object {
              "backgroundImage": "https://tvlk.imgix.net/imageResource/2018/10/29/1540783692952-6cd5699c5b98cd71d8b7a429cf61e04a.jpeg?auto=compress%2Cformat&cs=srgb&fm=pjpg&ixlib=java-1.1.12&q=75",
              "cornerLabelText": null,
              "iconImage": null,
              "ribbonText": null,
              "subtitle": null,
              "title": null,
            },
            "id": "traveloka://flight/search/one_way/0/SGN/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "link": "traveloka://flight/search/one_way/0/SGN/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "linkType": "DEEPLINK",
            "style": Object {
              "properties": Object {
                "additionalTextColor": "#e91d1d",
                "subtitleColor": "#8f8f8f",
                "titleColor": "#1b1b1b",
              },
            },
            "type": "default",
          },
          Object {
            "attributes": Object {
              "backgroundImage": "https://tvlk.imgix.net/imageResource/2018/10/29/1540783837164-675f0484bb92a99bbceb95adc07a18df.jpeg?auto=compress%2Cformat&cs=srgb&fm=pjpg&ixlib=java-1.1.12&q=75",
              "cornerLabelText": null,
              "iconImage": null,
              "ribbonText": null,
              "subtitle": null,
              "title": null,
            },
            "id": "traveloka://flight/search/one_way/0/CXR/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "link": "traveloka://flight/search/one_way/0/CXR/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "linkType": "DEEPLINK",
            "style": Object {
              "properties": Object {
                "additionalTextColor": "#e91d1d",
                "subtitleColor": "#8f8f8f",
                "titleColor": "#1b1b1b",
              },
            },
            "type": "default",
          },
          Object {
            "attributes": Object {
              "backgroundImage": "https://tvlk.imgix.net/imageResource/2018/10/29/1540783844147-113e0093539abc6d9fd9f25ad29c3184.jpeg?auto=compress%2Cformat&cs=srgb&fm=pjpg&ixlib=java-1.1.12&q=75",
              "cornerLabelText": null,
              "iconImage": null,
              "ribbonText": null,
              "subtitle": null,
              "title": null,
            },
            "id": "traveloka://flight/search/one_way/0/BKK/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "link": "traveloka://flight/search/one_way/0/BKK/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "linkType": "DEEPLINK",
            "style": Object {
              "properties": Object {
                "overlay": true,
                "subtitleColor": "#000000",
                "textHorizontalAlignment": "LEFT",
                "textPlacement": "OUTSIDE",
                "textVerticalAlignment": "BOTTOM",
                "titleColor": "#000000",
                "titleType": "TYPE_1",
              },
            },
            "type": "default",
          },
          Object {
            "attributes": Object {
              "backgroundImage": "https://tvlk.imgix.net/imageResource/2018/10/29/1540783851606-fb7354fe8012e6431fe43d17a596e873.jpeg?auto=compress%2Cformat&cs=srgb&fm=pjpg&ixlib=java-1.1.12&q=75",
              "cornerLabelText": null,
              "iconImage": null,
              "ribbonText": null,
              "subtitle": null,
              "title": null,
            },
            "id": "traveloka://flight/search/one_way/0/VII/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "link": "traveloka://flight/search/one_way/0/VII/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "linkType": "DEEPLINK",
            "style": Object {
              "properties": Object {
                "overlay": true,
                "subtitleColor": "#000000",
                "textHorizontalAlignment": "LEFT",
                "textPlacement": "OUTSIDE",
                "textVerticalAlignment": "BOTTOM",
                "titleColor": "#000000",
                "titleType": "TYPE_1",
              },
            },
            "type": "default",
          },
          Object {
            "attributes": Object {
              "backgroundImage": "https://tvlk.imgix.net/imageResource/2018/10/29/1540783861349-f18e45182731d3c56cbcfb403514e3c3.jpeg?auto=compress%2Cformat&cs=srgb&fm=pjpg&ixlib=java-1.1.12&q=75",
              "cornerLabelText": null,
              "iconImage": null,
              "ribbonText": null,
              "subtitle": null,
              "title": null,
            },
            "id": "traveloka://flight/search/one_way/0/HPH/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "link": "traveloka://flight/search/one_way/0/HPH/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
            "linkType": "DEEPLINK",
            "style": Object {
              "properties": Object {
                "overlay": true,
                "subtitleColor": "#000000",
                "textHorizontalAlignment": "LEFT",
                "textPlacement": "OUTSIDE",
                "textVerticalAlignment": "BOTTOM",
                "titleColor": "#000000",
                "titleType": "TYPE_1",
              },
            },
            "type": "default",
          },
        ],
        "link": null,
        "sectionId": "vi-best-flight-deal",
        "sectionName": "vi-best-flight-deal",
        "style": Object {
          "properties": Object {
            "ratio": "3:4",
            "visibleItems": "1.8",
          },
          "subtitleColor": "#696969",
          "titleColor": "#000000",
          "type": "IMAGE_CAROUSEL",
        },
        "subtitle": "Big savings on flights! Book now!",
        "title": "Fly home for cheap this Tet Holiday",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
